src/cli.js

```diff
@@ -3,30 +3,64 @@ import {relative, basename, dirname, join} from 'path';
 import mkdirp from 'mkdirp';
 import controller from './cli-controller';
 import usage from './usage';
+import fragmentFilesForDocument from './fragment-files-for-document';
 
-function read(path) {
-  // Double space after read to align with [WRITE]
-  console.log(`[READ]  ${path}`);
+function read(path, silent) {
+  if (!silent) {
+    // Double space after read to align with [WRITE]
+    console.log(`[READ]  ${path}`);
+  }
 
   return readFileSync(path).toString();
 }
 
-function write(path, body) {
-  console.log(`[WRITE] ${path}`);
+function write(path, body, silent) {
+  if (!silent) {
+    console.log(`[WRITE] ${path}`);
+  }
 
   return writeFileSync(path, body);
 }
 
-function readDocuments(documents) {
+function readDocuments(documents, silent) {
   return documents.map((path) => {
     return {
-      body: read(path),
+      body: read(path, silent),
       path
     };
   });
 }
 
-function compileDocuments({documentCode, documentCompiler, outdir}) {
+function concatBodies(files) {
+  return files.reduce((buffer, file) => {
+    return buffer + file.body;
+  }, '');
+}
+
+function concatenateAndMarkFragments(documentCode) {
+  return documentCode.map((document) => {
+    const fragmentFiles = fragmentFilesForDocument(document.path, document.body);
+
+    if (fragmentFiles.length) {
+      const fragments = documentCode.filter((possibleFragment) => {
+        return fragmentFiles.includes(possibleFragment.path);
+      });
+
+      fragments.forEach((fragment) => {
+        fragment.isFragment = true;
+      });
```

Comment:
Only part I found weird was returning documents here but also relying on the side effect of `isFragment` being set. Not sure if there's a better way around it though.